### ISSUE TITLE
URI.DisableExternalResources in conjunction with HTML.SafeIframe

### ISSUE THREAD

**Issue by doMynation** [ezyang/htmlpurifier#247]
Hi,

We're currently using `HTML.SafeIframe` (along with `URI.SafeIframeRegexp`) to allow/whitelist YouTube and Vimeo videos and everything works well.

That said, we now need to also allow `<img>` tags, and only for images hosted on our domain. We've tried using `URI.DisableExternalResources` with `URI.Host` set to our domain, which works on its own for images, but causes an issue for other kinds of external resources (e.g. iframes). Given that `URI.Host` is set to our domain and given that it only accepts one domain, restricting external resources to our domain means we disable Youtube/Video videos. 

In other words, it does seem like `URI.DisableExternalResources` trumps `HTML.SafeIframe`, which means it's not possible to disable external images while also allowing iframes for Youtube.

Is there a workaround for that by any chance?

Thank you.

**Comment by ezyang** [comment]
Instead of DisableExternalResources, why don't you use the other option that lets you whitelist what allowed domains are? Then you can allow local, youtube and vimeo and disable everything else.

**Comment by doMynation** [comment]
I'm not sure which other option you're referring to. I did check again, and I couldn't find an option to white-list multiple domains. Were you referring to `URI.Host`? If so, it only accepts one domain.

**Comment by ezyang** [comment]
You're right. Probably the easiest way to get this functionality then is to make a custom URIFilter. You can look at the existing URIFilters in  library/HTMLPurifier/URIFilter to get an idea about how you can go about writing them. It will probably be some mixture of how SafeIframe.php is implemented along with DisableExternalResources.php. Use getURIDefinition on the Config object to get the object you can register them with.

If you can come up with a PR that adds this to core I'd probably accept it.
